pipeline-scripts/release.groovy

```diff
@@ -501,29 +501,34 @@ def stagePublishClient(quay_url, from_release_tag, release_name, arch, client_ty
     def CLIENT_MIRROR_DIR="${BASE_TO_MIRROR_DIR}/${arch}/clients/${client_type}/${release_name}"
     sh "mkdir -p ${CLIENT_MIRROR_DIR}"
 
+    def tools_extract_cmd = "MOBY_DISABLE_PIGZ=true GOTRACEBACK=all oc adm release extract --tools --command-os='*' -n ocp " +
+                                " --to=${CLIENT_MIRROR_DIR} --from ${quay_url}:${from_release_tag}"
+    commonlib.shell(script: tools_extract_cmd)
+
     if ( arch == 'x86_64' ) {
+        // oc_mirror_extract_cmd needs to extract to a emtpy folder, since tools_extract_cmd is already extracting to
+        // CLIENT_MIRROR_DIR, creating a temporary folder
+        def TEMP_OC_MIRROR_DIR = "${CLIENT_MIRROR_DIR}/temp-oc-mirror-dir"
+        sh "mkdir ${TEMP_OC_MIRROR_DIR}"
+
         // oc image  extract requires an empty destination directory. So do this before extracting tools.
-        // oc adm release extract --tools does not require an empty directory.
+        // oc adm release extract --tools does not require an empty directory, but will overwrite the files.
         def oc_mirror_extract_cmd = '''
             # If the release payload contains an oc-mirror artifact image, then extract the oc-mirror binary.
             if oc adm release info ${QUAY_URL}:${FROM_RELEASE_TAG} --image-for=oc-mirror ; then
-                MOBY_DISABLE_PIGZ=true GOTRACEBACK=all oc image extract `oc adm release info ${QUAY_URL}:${FROM_RELEASE_TAG} --image-for=oc-mirror` --path /usr/bin/oc-mirror:${CLIENT_MIRROR_DIR}
+                MOBY_DISABLE_PIGZ=true GOTRACEBACK=all oc image extract `oc adm release info ${QUAY_URL}:${FROM_RELEASE_TAG} --image-for=oc-mirror` --path /usr/bin/oc-mirror:${TEMP_OC_MIRROR_DIR}
                 pushd ${CLIENT_MIRROR_DIR}
-                tar zcvf oc-mirror.tar.gz oc-mirror
+                tar zcvf oc-mirror.tar.gz ${TEMP_OC_MIRROR_DIR}/oc-mirror
```

Comment:
This will include the temp dir in the tarball (you can use `tar -tzvf oc-mirror.tar.gz` to check the content in the tarball).
You can try `--confirm` option to extract to a non-empty directory, or create a tarball inside of `TEMP_OC_MIRROR_DIR`.

Reply:
I'll go with option B then, create a tarball inside `TEMP_OC_MIRROR_DIR`!

Reply:
Have updated the PR